### PR TITLE
Add loongarch linux-user emulation support

### PR DESCRIPTION
**Pull request by gaosong-loongson**
Base on:
https://patchew.org/QEMU/1630586467-22463-1-git-send-email-gaosong@loongson.cn/

### CODE REVIEW

**Comment by gaosong-loongson** [comment]
@yetist 

按照https://github.com/loongson/qemu/pull/2#issuecomment-915242170 操作，
能够正常编译和运行。 但会有很多告警。

告警：
...
/usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LVL45':
(.text+0x294): 警告：Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LM41':
(.text+0x128): 警告：Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LVL20':
(.text+0x158): 警告：Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
...

**Comment by yetist** [comment]
> @yetist
> 
> 按照[#2 (comment)](https://github.com/loongson/qemu/pull/2#issuecomment-915242170) 操作，
> 能够正常编译和运行。 但会有很多告警。
> 
> 告警：
> ...
> /usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LVL45': (.text+0x294): 警告：Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking /usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LM41':
> (.text+0x128): 警告：Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
> /usr/bin/ld: /lib/loongarch64-linux-gnu/libglib-2.0.a(gutils.c.o): in function `.LVL20':
> (.text+0x158): 警告：Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
> ...

这个警告跟 patch 无法，是 libglib-2.0.a 的问题，你可以编译个动态链接的试下。

File: linux-user/loongarch64/target_syscall.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    target_ulong csr_euen;
+    target_ulong csr_ecfg;
+    target_ulong csr_estat;
+    target_ulong csr_era;
+    target_ulong csr_badvaddr;
+    target_ulong orig_a0;
+    target_ulong __last[0];
+};
+
+#define UNAME_MACHINE "loongarch"
```
Comment (sunhaiyong1978) [inline linux-user/loongarch64/target_syscall.h:31]: 此处建议改为"loongarch64"，保持和真机一致。

Comment (gaosong-loongson) [reply linux-user/loongarch64/target_syscall.h]: 已修改。

**Pull request merged**